esmf_regrid/experimental/io.py

```diff
@@ -43,6 +57,49 @@ def _add_mask_to_cube(mask, cube, name):
         cube.add_aux_coord(mask_coord, list(range(cube.ndim)))
 
 
+@contextmanager
+def _managed_var_name(src_cube, tgt_cube):
+    src_coord_names = []
+    src_mesh_coords = []
+    if src_cube.mesh is not None:
+        src_mesh = src_cube.mesh
+        src_mesh_coords = src_mesh.coords()
+        for coord in src_mesh_coords:
+            src_coord_names.append(coord.var_name)
+    tgt_coord_names = []
+    tgt_mesh_coords = []
+    if tgt_cube.mesh is not None:
+        tgt_mesh = tgt_cube.mesh
+        tgt_mesh_coords = tgt_mesh.coords()
+        for coord in tgt_mesh_coords:
+            tgt_coord_names.append(coord.var_name)
+
+    try:
+        for coord in src_mesh_coords:
+            coord.var_name = "_".join([SOURCE_NAME, "mesh", coord.name()])
+        for coord in tgt_mesh_coords:
+            coord.var_name = "_".join([TARGET_NAME, "mesh", coord.name()])
+        yield None
+    finally:
+        for coord, var_name in zip(src_mesh_coords, src_coord_names):
+            coord.var_name = var_name
+        for coord, var_name in zip(tgt_mesh_coords, tgt_coord_names):
+            coord.var_name = var_name
+
+
+def _clean_var_names(cube):
```

Comment:
Again, because this takes a cube + returns a cube, it gives the impression that it creates + returns a new cube, leaving the original argument alone.  I.E. it appears to have a functional style.
But in fact, it makes _permanent_ changes to the argument cube, and then just returns that.

I think that, if you _want_ to create a new cube (and leave the argument alone), then you should _copy_ it. 
Or if not, you can just define and use this routine as modifying its argument (as also suggested above for `_managed_var_name` ) : no need to return a cube.

Reply:
The problem with copying a cube is that we've set up cube copying in such a way that the mesh on a copy is the same mesh as the original, so if you edit the mesh on the copy of a cube you are editing the mesh on the orignal cube. Also, this function is only called on the loaded cube so editing it in place makes more sense, I can change the return line though.